docs/content/sips/012-signing-spin-releases.md

```diff
@@ -0,0 +1,206 @@
+However, this has a few disadvantages, in particular related to key management
+(both from the perspective of the Spin project, which would need to carefully
+manage a private key, and for users who need to get the public key). In general,
+using this mechanism to sign software has proven to be difficult to convince users
+to adopt because of the complexity around GPG.
+- using Cosign with keys — this would still require the Spin project to maintain
+a private key
+
+The keyless mode of Cosign can eliminate the key management issue through
+ephemeral keys, which is why this method is preferred to alternatives.
+
+### Security considerations
+
+[Sigstore's trust model](https://docs.sigstore.dev/security/) relies on a few
+key trusted components: the OIDC provider to prove identity, the trust root
+based on [TUF](https://theupdateframework.io/), [Rekor](https://docs.sigstore.dev/rekor/overview/),
+and [Fulcio](https://docs.sigstore.dev/fulcio/overview/).
+
+If any of the components above are compromised, the security guarantees of _any_
+signature is no longer valid. However, because of the transparency logs,
+such compromises can be detected.
+
+Unauthorized access to the GitHub actions used to generate the releases could
+allow an attacker to tamper with the release — however, that attack surface is
+present regardless of what signature mechanism we choose — which suggests increased
+attention to the workflows used to build and generate release artifacts.
+
+### Appendix: minimal example in GitHub Actions
+
+[The following repository](https://github.com/radu-matei/keyless-cosign-demo) can
+be used as a minimal example for this workflow:
+
+```yml
+jobs:
+  sign:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write 
+
+    name: Sign artifact and publish signature and certificate
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@main
+        with:
+          cosign-release: v2.0.0
+
+      - name: Create an artifact
+        run: |
+          mkdir out
+          echo 'hello world' > out/artifact
+          
+      - name: Sign the artifact with GitHub OIDC token
+        run: cosign sign-blob --output-certificate out/crt.pem --output-signature out/artifact.sig out/artifact --yes
+
+      - name: Upload assets as GitHub artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: artifact
+          path: out/*
+      ...(continue with adding artifacts to the release)
+```
+
+After the action is executed, the new artifact is released and its signature
+published using Sigstore. To validate the signature, a user will then need to
+download the release package (which now contains the artifact itself, the
+certificate used to sign, and the signature itself), then validate the signature
+using `cosign`:
+
+```bash
+$ tree .
+├── artifact
+├── artifact.sig
+└── crt.pem
+
+$ cosign verify-blob \
```

Comment:
For me, it would be super useful to have the user experience of this clearly called out - the "GitHub actions example" is not a natural place to look for it _grin_

Reply:
Users should have this in a documentation section on the developer website.
For the SIP, I renamed the section.